File: articles/r-biological-data-analysis-visualization/python/metadata_manifest.py
"""
Metadata and data dictionary summary.

Run:
    python python/metadata_manifest.py
"""

from pathlib import Path

import pandas as pd


ARTICLE_DIR = Path(__file__).resolve().parents[1]


def main() -> None:
    dictionary = pd.read_csv(ARTICLE_DIR / "data" / "data_dictionary.csv")

    summary = pd.DataFrame(
        {
            "n_variables": [len(dictionary)],
            "n_required": [(dictionary["required"].astype(str).str.lower() == "true").sum()],
            "n_with_units": [dictionary["unit"].notna().sum()],
            "n_with_allowed_values": [dictionary["allowed_values"].notna().sum()],
        }
    )

    print(summary.to_string(index=False))
    print(dictionary.to_string(index=False))


if __name__ == "__main__":
    main()

File: articles/data-measurement-reproducibility-life-sciences/r/metadata_manifest.R
# Metadata manifest workflow.

dict_path <- file.path("data", "data_dictionary.csv")
artifact_path <- file.path("data", "artifact_manifest.csv")

if (!file.exists(dict_path)) {
  dict_path <- file.path("..", "data", "data_dictionary.csv")
  artifact_path <- file.path("..", "data", "artifact_manifest.csv")
}

dictionary <- read.csv(dict_path)
artifacts <- read.csv(artifact_path)

dictionary_summary <- data.frame(
  n_variables = nrow(dictionary),
  n_required = sum(tolower(dictionary$required) == "true"),
  n_with_units = sum(dictionary$unit != "")
)

artifact_summary <- aggregate(
  artifact_name ~ artifact_role,
  data = artifacts,
  FUN = length
)

names(artifact_summary)[names(artifact_summary) == "artifact_name"] <- "n_artifacts"

print(dictionary_summary)
print(artifact_summary)
